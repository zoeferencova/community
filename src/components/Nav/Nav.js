import React, { Component } from "react";
import { NavLink } from "react-router-dom";
import { PropTypes } from 'prop-types';
import CommUnityContext from "../../contexts/context";
import { ProfilePicture } from "../Utils/Utils";

import styles from "./Nav.module.css";

export default class Nav extends Component {
    static contextType = CommUnityContext;

    handleClick(e) {
        return (window.location.pathname === "/location" && !this.context.user.location.lat) ? this.props.isLoggedIn && e.preventDefault() : ""
    }

    render() {
        return (   
            <nav className={styles.nav}>
                {this.props.isLoggedIn ?
                    <NavLink to={"/home"} onClick={e => this.handleClick(e)}><img src={require("../../images/small-logo.png")} alt="logo" className={styles.smallLogo} /></NavLink>
                :
                    <NavLink to={"/"}><img src={require("../../images/big-logo.png")} alt="logo" className={styles.bigLogo} /></NavLink>
                }
                {this.props.isLoggedIn ? 
                    <>
                        <ul className={`${styles.navLinks} ${styles.mobile} navLinks`}>
                            <NavLink to="/messages" onClick={e => this.handleClick(e)}><li><i className="fas fa-comment"></i></li></NavLink>
                            <NavLink to="/account" onClick={e => this.handleClick(e)}><li><ProfilePicture first_name={this.props.first_name} /></li></NavLink>
                        </ul>
                        <ul className={`${styles.navLinks} ${styles.desktop}`}>
                            <NavLink to="/home" onClick={e => this.handleClick(e)}><li><i className="fas fa-home"></i></li></NavLink>
                            <NavLink to="/messages" onClick={e => this.handleClick(e)}><li><i className="fas fa-comment"></i></li></NavLink>
                            <NavLink to="/account" onClick={e => this.handleClick(e)}><li><ProfilePicture className={styles.profPic} first_name={this.props.first_name} /></li></NavLink>
                        </ul>
                    </>
                :
                    <ul className={styles.navLinks}>
                        <NavLink to="/login"><li className={styles.logIn}>Log in</li></NavLink>
                        <NavLink to="/register"><li className={styles.signUp}>Sign up</li></NavLink>
                    </ul>
                } 
            </nav>  
        )
    }
}

Nav.propTypes = {
    isLoggedIn: PropTypes.bool,
    first_name: PropTypes.string
}