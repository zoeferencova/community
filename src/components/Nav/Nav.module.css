.nav {
    display: flex;
    flex-direction: row;
    justify-content: space-between;
    align-items: center;
    z-index: 100;
    height: 65px;
    position: relative;
    width: 100%;
    min-height: 80px;
}

.bigLogo {
    height: 90px;
    margin-left: 50px;
    margin-top: 60px;
}

.smallLogo {
    width: 90px;
    align-self: center;
    margin-top: 5px;
    margin-left: 20px;
}

.navLinks {
    list-style: none;
    display: flex;
    font-size: 15px;
    letter-spacing: 0.7px;
    margin-right: 35px;
}

.navLinks li {
    margin-right: 20px;
}

.navLinks i {
    font-size: 25px;
    margin-right: 10px;
}

.navLinks a {
    text-decoration: none;
    color: #96a5b3;
    align-self: center;
}

:global .active li {
    color: #435361;
    transition: 0.5s;
}


.profPic {
    align-self: center;
}

.logIn, .signUp {
    border: 1px solid black;
    border-radius: 3px;
    padding: 10px 12px !important;
}

.logIn {
    color: black;
}

.signUp {
    background-color: black;
    color: #ebebeb;
}

.mobile {
    display: none;
}

@media only screen and (max-width: 500px) {
    .logIn, .signUp {
        font-size: 12.5px;
        padding: 8px 10px;
    }

    .mobile {
        display: flex;
    }

    .desktop {
        display: none;
    }

    :global .active li {
        border: none;
    }

    .navLinks {
        margin-right: 10px;
    }

    .smallLogo {
        margin-left: 0px;
        width: 80px;
    }
}

