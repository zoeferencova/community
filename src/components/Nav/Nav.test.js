import React from "react";
import { render } from "enzyme";
import Nav from "./Nav";
import { BrowserRouter } from "react-router-dom";

describe("Nav", () => {
  it("renders Nav component when not logged in", () => {
    const mockProps = {
        isLoggedIn: false
    }

    const wrapper = render(
      <BrowserRouter>
        <Nav {...mockProps} />
      </BrowserRouter>
    );
    
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find("#loggedOut").length).toBe(2);
    expect(wrapper.find("#loggedIn").length).toBe(0);
  });

  it("renders Nav component when logged in", () => {
    const mockProps = {
        isLoggedIn: true,
        first_name: "Zoe"
    }

    const wrapper = render(
      <BrowserRouter>
        <Nav {...mockProps} />
      </BrowserRouter>
    );
    
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find("#loggedIn").length).toBe(2);
    expect(wrapper.find("#loggedOut").length).toBe(0);
  });
});



