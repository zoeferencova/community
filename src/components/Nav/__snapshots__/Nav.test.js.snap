// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav renders Nav component when logged in 1`] = `
<nav
  class="nav"
>
  <a
    href="/home"
    id="loggedIn"
  >
    <img
      alt="logo"
      class="smallLogo"
      src="small-logo.png"
    />
  </a>
  <div
    id="loggedIn"
  >
    <ul
      class="navLinks mobile navLinks"
    >
      <a
        href="/messages"
      >
        <li>
          <i
            class="fas fa-comment"
          />
        </li>
      </a>
      <a
        href="/account"
      >
        <li>
          <div
            class="circle blue "
            first_name="Zoe"
          >
            Z
          </div>
        </li>
      </a>
    </ul>
    <ul
      class="navLinks desktop"
    >
      <a
        href="/home"
      >
        <li>
          <i
            class="fas fa-home"
          />
        </li>
      </a>
      <a
        href="/messages"
      >
        <li>
          <i
            class="fas fa-comment"
          />
        </li>
      </a>
      <a
        href="/account"
      >
        <li>
          <div
            class="circle blue profPic"
            first_name="Zoe"
          >
            Z
          </div>
        </li>
      </a>
    </ul>
  </div>
</nav>
`;

exports[`Nav renders Nav component when not logged in 1`] = `
<nav
  class="nav"
>
  <a
    aria-current="page"
    class="active"
    href="/"
    id="loggedOut"
  >
    <img
      alt="logo"
      class="bigLogo"
      src="big-logo.png"
    />
  </a>
  <ul
    class="navLinks"
    id="loggedOut"
  >
    <a
      href="/login"
    >
      <li
        class="logIn"
      >
        Log in
      </li>
    </a>
    <a
      href="/register"
    >
      <li
        class="signUp"
      >
        Sign up
      </li>
    </a>
  </ul>
</nav>
`;
